services/management/adapter/file_provider.go

```diff
@@ -78,15 +78,21 @@ func (mp *FileProvider) generatePath(referenceTime primitives.TimestampSeconds)
 }
 
 func (mp *FileProvider) readUrl(path string) ([]byte, error) {
-	res, err := http.Get(path)
+	res, err := mp.client.Get(path)
 
 	if err != nil || res == nil {
 		return nil, errors.Wrapf(err, "Failed http get of url %s", path)
 	} else if res.ContentLength > 0 && uint32(res.ContentLength) > mp.config.ManagementMaxFileSize() { // TODO when no length given find other way ?
 		return nil, errors.Wrapf(err, "Failed http get response too big %d", res.ContentLength)
 	}
 
-	return ioutil.ReadAll(res.Body)
+	contents, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not read http body")
+	}
+	defer res.Body.Close()
```

Comment:
We want to do close the Body if `res.Body != null`, right now if you had an error reading the body you will not close it